sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -1479,3 +1479,20 @@ def test_issue_8653():
 def test_issue_9157():
     n = Symbol('n', integer=True, positive=True)
     atan(n - 1).is_nonnegative is True
+
+
+def test_trig_period():
+    x, y = symbols('x, y')
+
+    assert sin(x).period() == 2*pi
+    assert cos(x).period() == 2*pi
+    assert tan(x).period() == pi
+    assert cot(x).period() == pi
+    assert sec(x).period() == 2*pi
+    assert csc(x).period() == 2*pi
+    assert sin(2*x).period() == pi
+    assert cot(4*x - 6).period() == pi/4
+    assert cos((-3)*x).period() == 2*pi/3
+    assert cos(x*y).period(x) == 2*pi/abs(y)
+    assert sin(3*x*y + 2*pi).period(y) == 2*pi/abs(3*x)
```

Comment:
[wolfram](https://www.wolframalpha.com/input/?i=period+y%3D+sin(3*x*y+%2B+2*pi)) returns `-2*pi/(3*x)`. Why not same ans ?